### DISCUSSION TITLE
Documentation variables could be more diverse?

### DISCUSSION THREAD

**Discussion by benoittgt** [stimulusjs/stimulus#819] [General]
Hello

This subject has been initially open [on discourse](https://discuss.hotwired.dev/t/documentation-variables-could-be-more-diverse/4767).  Feel free to close it if you want to keep this as a discussion.

I enjoy reading the documentation of Stimulus but sometimes I find it hard to understand how variables are passed when they are too similar. For example on “[outlet](https://stimulus.hotwired.dev/reference/outlets)” doc where in multiple case “result” could be different?

![](https://global.discourse-cdn.com/standard10/uploads/stimulusjs/original/2X/f/f90558f3927a300a17f79198f10eda0a63e5c845.png)

Same in [cross controller event](https://stimulus.hotwired.dev/reference/controllers#cross-controller-coordination-with-events). Where we don’t know what we can change at first sight and what needs to be similar. Like at first reading I was thinking ; what “copy” represent?

![](https://global.discourse-cdn.com/standard10/uploads/stimulusjs/original/2X/0/01fcbe39ddf139db03d683a3caa7884fc3c90894.png)

Also I rarely have variables with one word. Handling a mix of dash, underscore (Rails), cameCase can leads to easy mistakes. Maybe the doc could be more diverse on this part too in examples.

Thanks

**Comment by marcoroth** [comment]
Hey @benoittgt, thank you for taking the time to open this issue!

I really appreciate your feedback on the documentation and I agree with you that it could benefit from more diverse examples to better showcase the appropriate usage of each API. 

At the same time I think it's important that we don't use too different or too specific examples. So that they are still comparable and showcase how the API is used, without having to understand the context/domain of the example first, if that makes sense?

I'd be more than happy to merge any suggestions that you may have in mind to improve the documentation and provide a better understanding of the APIs. Thanks for bringing this up!
